cli/dcoscli/config/main.py

```diff
@@ -110,6 +110,75 @@ def _unset(name):
     return 0
 
 
+def _format_config(file_value, effective_value, name=None, envvar_name=None):
+    """
+    Construct a string to show on a terminal, indicating the value and
+    possibly other useful things such as the setting name and whether
+    it is being controlled by an environment variable.
+
+    >>> _format_config('x', 'x')
+    'x'
+    >>> _format_config('x', 'x', 'setting.name') ->
+    'setting.name x'
+    >>> _format_config('x', 'y', envvar_name='ENVVAR')
+    'y # overwritten by ENVVAR; config file value: x'
+    >>> _format_config('x', 'y', 'setting.name', envvar_name='ENVVAR')
+    'setting.name y   # overwritten by ENVVAR; config file value: x'
+
+    :param file_value: config value present in the toml file
+    :type file_value: str
+
+    :param effective_value: config value either from file or as overwritten
+                            from the environment
+    :type effective_value: str
+    :param name: config key (not used for single value show)
+    :type name: str|None
+    :param envvar_name: name of environment variable that overwote the value
+    :type envvar_name: str|None
+    :returns: formatted string for terminal
+    :raise ValueError: if overwritten and not envvar_name
+    :rtype: str
+    """
+
+    # when declaring that vars are overwritten by the environment,
+    # line up those messages to this column (unless the var name is long)
+    overwite_msg_display_column = 35
+
+    # When showing all values, don't print the token value;
+    if name == "core.dcos_acs_token":
+        print_value = "*"*8
+    else:
+        print_value = effective_value
+
+    if file_value == effective_value:
+        if name:
+            return '%s %s' % (name, print_value)
+        else:
+            return effective_value
+    else:
+        if not envvar_name:
+            exc_msg = "envvar_name must be provided if overwritten is true"
+            raise ValueError(exc_msg)
```

Comment:
This is a private function, so we have a little more leeway here. I'd prefer to just have tests that cover this case. What about returning "N/A" as envvar if its not given? This is a programmer error vs an API/network/error so would be covered by tests rather than in code. But at least N/A is still providing some feedback to the user.